### PR TITLE
Bugfix/df 629  candidate contact country selection

### PR DESCRIPTION
**Pull request by emina-str**

### CODE REVIEW

File: alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts [hunk 1]
```diff
@@ -30,4 +42,22 @@ export class IsoCountryService {
   get countryOptions$(): Observable<SelectableOption[]> {
     return this._countryOptions$;
   }
+
+  get countryOptionsSortedWithMainCountriesFirst$(): Observable<SelectableOption[]> {
```
Comment (pavax) [inline alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts]: unit test

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts [hunk 2]
```diff
@@ -143,7 +145,7 @@ export class ContactModalComponent extends AbstractSubscriber implements OnInit
       companyHouseNr: [null, [Validators.required, patternInputValidator(HOUSE_NUMBER_REGEX)]],
       companyZipCode: [null, Validators.required],
       companyCity: [null, Validators.required],
-      companyCountry: [null, Validators.required]
+      countryIsoCode: [null, Validators.required]
```
Comment (pavax) [inline alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: we must NOT send the ISO Code to the backend but rather the actual value (map it in the request-mapper)

File: alv-portal-ui/src/app/candidate-search/candidate-search.module.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ import { CandidateSearchGuard } from './candidate-search/candidate-search.guard'
 import { CandidateSearchFilterParameterService } from './candidate-search/candidate-search-filter-parameter.service';
 import { CandidateProfileBadgesMapperService } from './candidate-profile-badges-mapper.service';
 import { WidgetsModule } from '../widgets/widgets.module';
+import { IsoCountryService } from '../job-advertisement/job-publication/job-publication-form/iso-country.service';
```
Comment (pavax) [inline alv-portal-ui/src/app/candidate-search/candidate-search.module.ts]: move the IsoCountryService  to the shared module

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts [hunk 4]
```diff
@@ -177,16 +197,38 @@ export class ContactModalComponent extends AbstractSubscriber implements OnInit
...
+      city: company.companyCity,
+      country: company.country
     };
   }
 
+  private mapEmailContent(formValue: any): Observable<EmailContactModal> {
+
+    return this.countryOptions$.pipe(
+      map((countryOptions) => {
+        const countryIsoCode = this.form.get('company').get('countryIsoCode').value;
```
Comment (pavax) [inline alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: what happens if  this.form.get('company') return null?

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts [hunk 5]
```diff
@@ -177,16 +197,38 @@ export class ContactModalComponent extends AbstractSubscriber implements OnInit
...
+        const country = countryOptions.find(country => country.value === countryIsoCode);
+        const company = Object.assign({}, formValue.company);
+        company.country = country.label;
+        return {
+          candidateId: this.candidate.id,
+          company: this.mapCountry(company),
```
Comment (pavax) [inline alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: suggestion: the method mapCountry should handle anyting related to mapping to country so

`this.mapCountry(this.form.get('company'), countryOptions);`

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts [hunk 6]
```diff
@@ -177,16 +197,38 @@ export class ContactModalComponent extends AbstractSubscriber implements OnInit
     });
   }
 
-  private mapEmailContent(formValue: any): EmailContactModal {
+  private mapCountry(company: any): Company {
```
Comment (pavax) [inline alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: I guess that method should be called `mapCompany`

File: alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts [hunk 7]
```diff
@@ -28,6 +36,23 @@ export class IsoCountryService {
   }
 
   get countryOptions$(): Observable<SelectableOption[]> {
```
Comment (pavax) [inline alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts]: create a krama unit test for the countryOptions

File: alv-portal-ui/src/app/candidate-search/candidate-search.module.ts (hunk 3 above)
Comment (emina-str) [reply alv-portal-ui/src/app/candidate-search/candidate-search.module.ts]: done.

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts (hunk 4 above)
Comment (emina-str) [reply alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: Yes, we had that in mind, but didn't fix it at that point :). Now will be null-safe.

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts (hunk 5 above)
Comment (emina-str) [reply alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: fixed.

File: alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts (hunk 6 above)
Comment (emina-str) [reply alv-portal-ui/src/app/candidate-search/candidate-detail/contact-modal/contact-modal.component.ts]: yes

File: alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts (hunk 1 above)
Comment (emina-str) [reply alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts]: wrote test.

File: alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts (hunk 7 above)
Comment (emina-str) [reply alv-portal-ui/src/app/job-advertisement/job-publication/job-publication-form/iso-country.service.ts]: done.

File: alv-portal-ui/src/app/shared/shared.module.ts [hunk 8]
```diff
@@ -48,7 +49,8 @@ import { LegalTermsModalComponent } from './legal-terms/legal-terms-modal/legal-
     LegalTermsDirective
   ],
   providers: [
-    LandingPageGuard
+    LandingPageGuard,
+    IsoCountryService
```
Comment (pavax) [inline alv-portal-ui/src/app/shared/shared.module.ts]: the IsoCountryService .ts file should now also be moved to the shared directory

**Pull request merged**